### PR TITLE
Upload Windows SBOMs to python.org

### PR DESCRIPTION
**Pull request by sethmlarson**
I wasn't quite sure how to test these changes since we don't actually want to upload to python.org.

### CODE REVIEW

File: windows-release/stage-publish-pythonorg.yml [hunk 1]
```diff
@@ -57,6 +57,17 @@ jobs:
         buildVersionToDownload: specific
         buildId: ${{ parameters.BuildToPublish }}
 
+    - task: DownloadBuildArtifacts@0
+      displayName: 'Download artifact from ${{ parameters.BuildToPublish }}: sbom'
+      inputs:
+        artifactName: sbom
+        downloadPath: $(Build.BinariesDirectory)\sbom
```
Comment (sethmlarson) [inline windows-release/stage-publish-pythonorg.yml:64]: Does `DownloadBuildArtifact` create the directory at `downloadPath`?

File: windows-release/uploadrelease.ps1 [hunk 2]
```diff
@@ -87,19 +88,28 @@ $dirs = gci "$build" -Directory
...
     $msi = gci *.msi, *.msi.asc, *.msu, *.msu.asc
+    $spdx_json = gci *.spdx.json
     popd
 
     if ($exe) {
         & $pscp -batch -hostkey $hostkey -noagent -i $keyfile $exe.FullName "$user@${server}:$d"
         if (-not $?) { throw "Failed to upload $exe" }
     }
 
+    if ($spdx_json) {
```
Comment (sethmlarson) [inline windows-release/uploadrelease.ps1:108]: This probably isn't right, the SBOM directory contains more than one file, but I believe other artifacts are like that too so I duplicated what was done for exe?

File: windows-release/uploadrelease.ps1 [hunk 3]
```diff
@@ -87,19 +88,28 @@ $dirs = gci "$build" -Directory
 if ($embed) {
     $dirs = ($dirs, (gi $embed)) | %{ $_ }
 }
+if ($sbom) {
+    $dirs = ($dirs, $sbom)
```
Comment (zooba) [inline windows-release/uploadrelease.ps1:92]:
```suggestion
    $dirs = ($dirs, $sbom) | %{ $_ }
```

Theoretically, the last bit shouldn't be needed (it serves to flatten the two arrays), but I'm sure I put it there on line 88/89 for a reason.

File: windows-release/uploadrelease.ps1 [hunk 4]
```diff
@@ -87,19 +88,28 @@ $dirs = gci "$build" -Directory
...
     popd
 
     if ($exe) {
         & $pscp -batch -hostkey $hostkey -noagent -i $keyfile $exe.FullName "$user@${server}:$d"
         if (-not $?) { throw "Failed to upload $exe" }
     }
 
+    if ($spdx_json) {
+        & $pscp -batch -hostkey $hostkey -noagent -i $keyfile $spdx_json.FullName "$user@${server}:$d"
+        if (-not $?) { throw "Failed to upload $spdx_json" }
```
Comment (zooba) [inline windows-release/uploadrelease.ps1:110]:
```suggestion
        if (-not $?) { Write-Host "##[warning]Failed to upload $spdx_json" }
```

At least until we're confident this is going to work, I'd rather not break the next release, especially after files have already been uploaded. All a break here does is make it highly likely that we'll end up with a second build and mismatched binaries between sources (e.g. Nuget will have already published by this point).

**Review by zooba — COMMENTED** [review]
Two changes and I think this is good to merge. We can't really test until a release comes around, but I guess 3.13b1 isn't far off now.

**Comment by sethmlarson** [comment]
Thanks for the review @zooba! I applied your suggestions :)

**Comment by zooba** [comment]
Alright, we're in! Will find out what happens when we release next...

**Pull request merged**